bids-validator/validators/bids/checkConsistency.js

```diff
@@ -0,0 +1,228 @@
+
+    // Find the path before filename
+    let regExp = new RegExp('(.*/).*')
+    let preFilename = regExp.exec(possibleJsonPath)[1]
+
+    // Retrieve all json files at the same path with ome-tiff files
+    let jsonPaths = potentialSidecars.filter(path => {
+      let jsonpath = regExp.exec(path)[1]
+      return jsonpath === preFilename
+    })
+
+    // if possible json paths are not empty
+    jsonFiles.forEach(jsonFile => {
+      // if possible json file exists
+      if (jsonPaths.includes(jsonFile.relativePath)) {
+        components.push({
+          omeFile: omeFile,
+          jsonFilePath: jsonFile.relativePath,
+        })
+      }
+    })
+  })
+
+  return components
+}
+
+const checkConsistency = (omeFiles, jsonFiles, jsonContentsDict) => {
+  let issues = []
+
+  let components = getMatchingComponents(omeFiles, jsonFiles)
+
+  // if at least one ome-tiff file has no corresponding json file
+  if (components.length !== omeFiles.length) {
+    issues.push(new Issue({ code: 223 }))
+  }
+
+  const omeIssuesPromises = components.map(component => {
+    return utils.limit(
+      () =>
+        new Promise(async resolve => {
+          let jsonData = jsonContentsDict[component.jsonFilePath]
+          let omeData = await getOMETiffData(component.omeFile)
+          let optionalFieldsIssues = await validateOptionalFields(
+            omeData,
+            jsonData,
+          )
+          let pixelSizeIssues = await validatePixelSize(omeData, jsonData)
+          let matrixIssues = validateChunkTransformationMatrix(
+            component.jsonFilePath,
+            component.omeFile.path,
+            jsonData,
+          )
+          issues = issues
+            .concat(optionalFieldsIssues)
+            .concat(pixelSizeIssues)
+            .concat(matrixIssues)
+          return resolve()
+        }),
+    )
+  })
+
+  return new Promise(resolve =>
+    Promise.all(omeIssuesPromises).then(() => resolve(issues)),
+  )
+}
+
+const validateOptionalFields = async (omeData, jsonData) => {
+  let issues = []
+
+  let fields = {
+    Immersion: 'Immersion',
+    NumericalAperture: 'LensNA',
+    NominalMagnification: 'Magnification',
+  }
+
+  if (
+    omeData['OME']['Instrument'] &&
+    omeData['OME']['Instrument'][0]['Objective']
+  ) {
+    let objective = omeData['OME']['Instrument'][0]['Objective'][0]['$']
+    for (let field in fields) {
+      let property = fields[field]
+      if (jsonData.hasOwnProperty(field) && !objective[property]) {
+        issues.push(new Issue({ code: 225 }))
+      } else if (jsonData.hasOwnProperty(field) && objective[property]) {
+        if (objective[property] != jsonData[field]) {
+          issues.push(new Issue({ code: 226 }))
+        }
+      }
+    }
+  }
+
+  return issues
+}
+
+const validateChunkTransformationMatrix = (
+  jsonFilePath,
+  omeFilePath,
+  jsonData,
+) => {
+  let issues = []
+
+  /*if chunk-<index> is used either in the filenames of 
```

Comment:
We need to make this more general because PNG and TIF files can also have the` chunk-<index>` and the warning should also apply.

Reply:
Should we check PNG and TIF files only with microscopy?

Reply:
Yes, only for microscopy

Reply:
How about files with the `_photo` suffix? Should we ignore this type of file?

Reply:
The template for `_photo` suffix files does not include `chunk-<index>` so I guess they are going to be ignored anyway?
(https://github.com/neuropoly/bids-validator/blob/958bbeacce07468618461662f995c7ebb47b7fc6/bids-validator/bids_validator/rules/file_level_rules.json#L558)
But yes, for this check, `_photo` files are to be ignored.

Reply:
Done